### DISCUSSION TITLE
Failed to load env - Cannot read properties of undefined (reading 'split')

### DISCUSSION THREAD

**Discussion by hckhanh** [vercel/next.js#35818] [Help]
### Verify canary release

- [X] I verified that the issue exists in Next.js canary release

### Provide environment information

```sh
  Operating System:
      Platform: win32
      Arch: x64
      Version: Windows 10 Pro
    Binaries:
      Node: 16.14.2
      npm: N/A
      Yarn: N/A
      pnpm: N/A
    Relevant packages:
      next: 12.1.4
      react: 17.0.2
      react-dom: 17.0.2
```

### What browser are you using? (if relevant)

Chrome 100.0.4896.60

### How are you deploying your application? (if relevant)

next start

### Describe the Bug

I get this error when building my project:

```sh
error - Failed to load env from C:\Users\user\Desktop\project\.env TypeError: Cannot read properties of undefined (reading 'split')
```

### Expected Behavior

Should be succeeded

### To Reproduce

Use .env file
Then run `next dev` or `next build`

**Comment by hckhanh** [comment]
I have to downgrade to `12.1.0`
This issue appeared from `12.1.1`

**Comment by icyJoseph** [comment]
Do you have anymore information about what you are trying to do? I just spinned up a project with `12.1.1` and this in `.env`:

```
SOME_ENV_VAR=021929202
```

And either `dev` or `build` work fine.

How does the `.env` file look like? (remove secrets!) - and how are you using the env variables?

**Reply by hckhanh** [reply]
Hi @icyJoseph,
I tried `12.1.1` with this .env and get the error:

```env
SIGN_IN_VALIDATION="^.+@domain\.com$"
NEXT_PUBLIC_CDN_DOMAIN="cdn.domain.com"
```

It throws an error even though I haven't used any vars at all.

I realized that `12.1.0` can load the env vars in `next.config.js`, but it doesn't since `12.1.1`

**Reply by icyJoseph** [reply]
## Solution

I found the solution here: https://github.com/motdotla/dotenv-expand/issues/65

You need to escape the `$` in `SIGN_IN_VALIDATION`:

```
SIGN_IN_VALIDATION="^.+@domain\\.com\$"
```

Just make sure, it still works as expected.

---

## Investigation

I see, that information was key, should've included that in the original report.

I found this commit https://github.com/vercel/next.js/pull/34765 that was included in `12.1.1`, where `dotenv-expand` was upgraded from `5.x.x` to `8.x.x`.

So the minimal reproduction is:

1. `yarn create next-app test-app`
2. `cd test-app`
3. `yarn add next@12.1.1 --exact`
4. Create `.env` and fill it with `SOME_ENV_VAR="^.+@domain\\.com$"`
5. `yarn build` outputs the error.

~~Now we must think, what part of the dry REGEX must be culprit. With this information you should be able to re-open a bug report, I think.~~

**Reply by honocoder** [reply]
You are a true saviour, thanks a lot! Got stuck a few days on a bug and finally saw the light thanks to your comment

**Reply by 90splayer** [reply]
Thank you!!!

**Comment by ghost** [comment]
Ya me estaba doliendo la cabeza. ¡Gracias!
